libs/core/langchain_core/vectorstores/base.py

```diff
@@ -99,10 +98,14 @@ def add_texts(
                     f"Got {len(metadatas)} metadatas and {len(texts_)} texts."
                 )
             metadatas_ = iter(metadatas) if metadatas else cycle([{}])
+            ids_: Iterator[Optional[str]] = iter(ids) if ids else cycle([None])
             docs = [
-                Document(page_content=text, metadata=metadata_)
-                for text, metadata_ in zip(texts, metadatas_)
+                Document(id=id_, page_content=text, metadata=metadata_)
+                for text, metadata_, id_ in zip(texts, metadatas_, ids_)
             ]
+            if ids is not None:
+                # For backward compatibility
+                kwargs["ids"] = ids
```

Comment:
Any chance that the `ids` was passed as an `iter` in any existing code? If so, we should probably pick ids from the `Document` object?

Reply:
If `ids` was an `iter` in existing code, it means that `add_documents` accepts it. So it shouldn't cause any problem ?

Reply:
If `ids` was passed as an `iter` in existing code, it means that `add_documents` accepts it. So it shouldn't cause any problem ?
We are really just reproducing the previous behavior of having `ids` in the kwargs, whatever its value or its type.

Reply:
(last question) can merge if OK

does the code not exhaust the iterable here:

```python
            docs = [
                Document(id=id_, page_content=text, metadata=metadata_)
                for text, metadata_, id_ in zip(texts, metadatas_, ids_)
            ]


            if ids is not None:
                # For backward compatibility
                kwargs["ids"] = ids # <-- is it not exhaustted?
```

Reply:
OOOH ids_ and id we're all good

Reply:
Good catch! This would indeed be a problem. For typing too.
I checked all implementations and actually none that implements `add_documents` uses ids as an iter.
So I believe it's fine ?